builtin/logical/consul/path_token.go

```diff
@@ -132,3 +163,25 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 
 	return s, nil
 }
+
+// Parses the provided string-like value as a semicolon-separated list of values.
+func parseSemicolonStringSlice(in interface{}) ([]string, error) {
+	rawString, ok := in.(string)
+	if ok && rawString == "" {
+		return []string{}, nil
+	}
+	var result []string
+	config := &mapstructure.DecoderConfig{
```

Comment:
This seems like a lot of machinery for `strings.Split(rawString, ";")`? Or am I missing something?

Reply:
So, the original PR actually had implemented a TypeSemicolonStringSlice (as you mentioned in another comment). Jim thought it might be better to move the logic out of parseutil and just have a function here because it's the only place in the code that currently needs semicolon-delimited input.

This is just lifting the logic from that into here. I agree this can be cut down significantly.

Reply:
+1 on using `strings.Split` here since we're not doing anything particularly fancy like decoding to a custom struct.

Reply:
Please let me know what you think of the update

Reply:
This has now changed quite a bit by using Austin's suggestion to receive the two parameters as TypeStringSlice. This lets a user input the parameter multiple times and automatically builds the slice out of those inputs to eliminate the second strings.Split, thus simplifying the parsing logic.